### PR TITLE
Add support for importing MCF files in RSI.

### PR DESCRIPTION
**Pull request by keyurva**
* For main DC, it simply copies the MCF to the output directory.
* For custom DC, it parses the MCF into triples and inserts them into the triples table.
   + Parser script moved from the [tools repo](https://github.com/datacommonsorg/tools/blob/ffee941bd2d30108097c17a855b5cec62af093e2/kg_util/mcf_parser.py#L4) to this repo.
   + The original script is used unchanged.
   + The script in the tools repo deleted in this PR: https://github.com/datacommonsorg/tools/pull/264
* Namespace prefixes are removed before triples are inserted into the DB. This also fixes a related bug where certain property references had namespace prefixes in the DB.
* Note that there is no validation done in RSI and assumes nodes in MCFs are resolved. All validations will be done with the DC import tool once it is integrated.
* TODO: Populate NL sentences with names, descriptions and utterances associated with SVs and topics.

### CODE REVIEW

File: simple/stats/mcf_importer.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      self.reporter.report_success()
+    except Exception as e:
+      self.reporter.report_failure(str(e))
+      raise e
+
+
+def _to_triple(parser_triple: list[str]) -> Triple:
+  [subject_id, predicate, value, value_type] = parser_triple
+  if value_type == _ID:
+    return Triple(subject_id, predicate, object_id=value)
```
Comment (pradh) [inline simple/stats/mcf_importer.py:69]: But we don't want subject_id right?

Comment (pradh) [reply simple/stats/mcf_importer.py:69]: We want to group-by subject_id, find `dcid` and then use that as subject_id?

File: simple/kg_util/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Libraries for working with Data Commons KG
```
Comment (pradh) [inline simple/kg_util/README.md:1]: To confirm, stuff in kg_util is unedited from tools repo?

**Review by pradh — COMMENTED** [review]
Thanks!

File: simple/kg_util/README.md (hunk 2 above)
Comment (keyurva) [reply simple/kg_util/README.md:1]: Yes.

File: simple/stats/mcf_importer.py (hunk 1 above)
Comment (keyurva) [reply simple/stats/mcf_importer.py:79]: Yes! Maintaining dcid values in a map and using them as subject_id now.

File: simple/stats/mcf_importer.py [hunk 3]
```diff
@@ -51,20 +52,29 @@ def do_import(self) -> None:
...
+        parser_triples: list[list[str]] = []
+        # DCID references
+        dcids: dict[str, str] = {}
+        for parser_triple in mcf_to_triples(self.input_fh.read_string_io()):
+          [subject_id, predicate, value, _] = parser_triple
+          if predicate == _DCID:
+            dcids[subject_id] = value
+          parser_triples.append(parser_triple)
+
+        triples = list(map(lambda x: _to_triple(x, dcids), parser_triples))
```
Comment (pradh) [inline simple/stats/mcf_importer.py:64]: maybe use a helper func? (given 3 levels of indent)

File: simple/stats/mcf_importer.py [hunk 4]
```diff
@@ -51,20 +52,29 @@ def do_import(self) -> None:
       if self.is_main_dc:
         self.output_fh.write_string(self.input_fh.read_string())
       else:
-        triples = list(
-            map(lambda x: _to_triple(x),
-                mcf_to_triples(self.input_fh.read_string_io())))
+        parser_triples: list[list[str]] = []
+        # DCID references
+        dcids: dict[str, str] = {}
```
Comment (pradh) [inline simple/stats/mcf_importer.py:57]: nit: local2dcid?

File: simple/stats/mcf_importer.py [hunk 5]
```diff
@@ -51,20 +52,29 @@ def do_import(self) -> None:
...
       self.reporter.report_success()
     except Exception as e:
       self.reporter.report_failure(str(e))
       raise e
 
 
-def _to_triple(parser_triple: list[str]) -> Triple:
+def _to_triple(parser_triple: list[str], dcids: dict[str, str]) -> Triple:
   [subject_id, predicate, value, value_type] = parser_triple
+  subject_id = dcids.get(subject_id, subject_id)
```
Comment (pradh) [inline simple/stats/mcf_importer.py:77]: If `dcid` is missing it should be reported as an error?

File: simple/tests/stats/test_data/runner/input/input_dir_driven/variables.mcf [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Node: dcid:some_var1
+typeOf: dcs:StatisticalVariable
```
Comment (pradh) [inline simple/tests/stats/test_data/runner/input/input_dir_driven/variables.mcf:2]: Can you do:

```
Node: v1
dcid: "some_var1"
```

for the first node?

File: simple/stats/mcf_importer.py (hunk 1 above)
Comment (pradh) [reply simple/stats/mcf_importer.py:79]: I'm slightly puzzled this change didn't affect the tests ?

**Review by pradh — COMMENTED** [review]
Thanks for the edits.

File: simple/stats/mcf_importer.py (hunk 1 above)
Comment (keyurva) [reply simple/stats/mcf_importer.py:87]: Chatted offline. It was due to namespaces being stripped before inserting into DB. The latter can be removed once we move to MCF / JSON-LD exclusively.

File: simple/tests/stats/test_data/runner/input/input_dir_driven/variables.mcf (hunk 6 above)
Comment (keyurva) [reply simple/tests/stats/test_data/runner/input/input_dir_driven/variables.mcf:3]: Done.

File: simple/stats/mcf_importer.py (hunk 3 above)
Comment (keyurva) [reply simple/stats/mcf_importer.py]: Done.

File: simple/stats/mcf_importer.py (hunk 4 above)
Comment (keyurva) [reply simple/stats/mcf_importer.py]: Done.

File: simple/stats/mcf_importer.py (hunk 5 above)
Comment (keyurva) [reply simple/stats/mcf_importer.py]: Done.

**Review by pradh — APPROVED** [review]
Thanks for the changes!

**Comment by keyurva** [comment]
Thanks for the review!

**Pull request merged**